### PR TITLE
xdc: allow GTPE2_CHANNEL to have IO_LOC_PAIRS

### PR DESCRIPTION
**Pull request by acomodi**
The GTPE2_CHANNEL does not have intermediate buffers to the IOPADs. Therefore, it is necessary to allow it to have IO_LOC_PAIRS attributes to generate correct placement constraints based on the IO signals locations.

Signed-off-by: Alessandro Comodi <acomodi@antmicro.com>

### CODE REVIEW

File: xdc-plugin/xdc.cc [hunk 1]
```diff
@@ -57,7 +57,8 @@ const std::unordered_map<std::string, std::vector<std::string>> supported_primit
   {"IBUF", {"IO_LOC_PAIRS", "IOSTANDARD"}},
   {"IOBUF", {"IO_LOC_PAIRS", "IOSTANDARD", "DRIVE", "SLEW", "IN_TERM"}},
   {"IOBUFDS", {"IO_LOC_PAIRS", "IOSTANDARD", "SLEW", "IN_TERM"}},
-  {"IBUFDS_GTE2", {"IO_LOC_PAIRS"}}};
+  {"IBUFDS_GTE2", {"IO_LOC_PAIRS"}},
+  {"GTPE2_CHANNEL", {"IO_LOC_PAIRS"}}};
```
Comment (litghost) [inline xdc-plugin/xdc.cc:61]: This is different than the other cases, please add a nice comment.  Otherwise looks ok.

**Review by tmichalak — CHANGES_REQUESTED** [review]
Can you add a small test for this?

File: xdc-plugin/xdc.cc (hunk 1 above)
Comment (acomodi) [reply xdc-plugin/xdc.cc:65]: Done

**Comment by acomodi** [comment]
> Can you add a small test for this?

@tmichalak Done, I have also added the IBUFDS_GTE2 test which was missing as well

**Comment by tmichalak** [comment]
> > Can you add a small test for this?
> 
> @tmichalak Done, I have also added the IBUFDS_GTE2 test which was missing as well

Thanks, now it all LGTM

**Pull request merged**